File: test_with_bean/src/main/java/study/spring/test_with_bean/School.java
package study.spring.test_with_bean;

import org.springframework.beans.factory.annotation.Value;

public class School {

  @Value("${application.school.name}")
  private String name;
  @Value("${application.school.location}")
  private String location;
  @Value("${application.school.age}")
  private int age;

  public String getName() {
    return name;
  }

  public String getLocation() {
    return location;
  }

  public int getAge() {
    return age;
  }
}

File: configuration-class-map/src/main/kotlin/study/spring/configurationclassmap/School.kt
package study.spring.configurationclassmap

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConfigurationProperties(value = "school.my")
class Schools (
    val elementrySchool: School,
    val middleSchool: School,
    val highSchool: School,
)

data class School(
    val name: String,
    val age: Int,
)